src/builder.rs

```diff
@@ -107,6 +123,15 @@ impl UffdBuilder {
         if self.non_blocking {
             flags |= libc::O_NONBLOCK;
         }
+
+        if self.user_mode_only
+            && linux_version::linux_kernel_version()
+                .map(|v| v >= linux_version::Version::new(5, 11, 0))
+                .unwrap_or(true)
+        {
+            flags |= raw::UFFD_USER_MODE_ONLY as i32;
+        }
```

Comment:
I'm not sure I understood your reply in the PR thread - does adding this flag cause an EINVAL on pre-5.11 kernels?

If so please add a comment above this check so we know why we are performing a runtime check

Reply:
I just tested with kernel 4.9 and calling `userfaultfd` with `UFFD_USER_MODE_ONLY` sets errno to EINVAL. I found a more robust way to check was to always try opening the fd with `UFFD_USER_MODE_ONLY` but retry without if it gives EINVAL (here `linux_version` seems to be parsing `uname` which could theoretically be inaccurate if running on a weirdly named kernel).

Reply:
what does `uname -r` report on this machine?